### PR TITLE
Use Dekorate in the Spring Boot example and add documentation

### PR DESCRIPTION
**Pull request by Sgitario**
These changes are only about adding documentation for the Spring Boot example.
Out of scope:
- I didn't do any improvements in the framework as I would like to understand better where we are. 
- I will document the Quarkus example later in a separate pull request. 
- Add the steps from the documentation to be done automatically in the CI

### CODE REVIEW

File: samples/quarkus/src/main/java/io/javaoperatorsdk/webhook/admission/sample/quarkus/AdmissionControllerConfig.java [hunk 1]
```diff
@@ -42,7 +42,9 @@ public AdmissionController<Pod> mutatingController() {
   @Named(VALIDATING_CONTROLLER)
   public AdmissionController<Pod> validatingController() {
     return new AdmissionController<>((resource, operation) -> {
-      if (resource.getMetadata().getLabels().get(APP_NAME_LABEL_KEY) == null) {
+      if (resource.getMetadata() == null
```
Comment (csviri) [inline samples/quarkus/src/main/java/io/javaoperatorsdk/webhook/admission/sample/quarkus/AdmissionControllerConfig.java:45]: The metadata actually never can be null. Maybe this check.

File: samples/spring-boot/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:11
```
Comment (csviri) [inline samples/spring-boot/Dockerfile:1]: Wouldn't be better to build with jib?

**Review by csviri — APPROVED** [review]
Made some minor comments otherwise LGTM

File: samples/quarkus/src/main/java/io/javaoperatorsdk/webhook/admission/sample/quarkus/AdmissionControllerConfig.java (hunk 1 above)
Comment (Sgitario) [reply samples/quarkus/src/main/java/io/javaoperatorsdk/webhook/admission/sample/quarkus/AdmissionControllerConfig.java]: Yes, that was too much. Removed.

File: samples/spring-boot/Dockerfile (hunk 2 above)
Comment (csviri) [reply samples/spring-boot/Dockerfile:1]: I would perfer jib, it is much nicer solution imo.

Comment (csviri) [reply samples/spring-boot/Dockerfile:1]: Also it is used on other projects.

Comment (Sgitario) [reply samples/spring-boot/Dockerfile]: Changed.

**Review by csviri — APPROVED** [review]
LGTM

**Pull request merged**